src/componets/forms/FormFields.tsx

```diff
@@ -0,0 +1,224 @@
+import type { FormValues } from '../../utils/validation';
+import type { Country } from '../../types';
+import type {
+  FieldErrors,
+  UseFormRegister,
+  UseFormWatch,
+} from 'react-hook-form';
+import { useState } from 'react';
+import { useAppSelector } from '../../store/hooks';
+import { getPasswordStrength } from '../../utils/passwordStrength';
+
+interface FormFieldsProps {
+  register?: UseFormRegister<FormValues>;
+  errors?: FieldErrors<FormValues>;
+  watch?: UseFormWatch<FormValues>;
+  countries: Country[];
+  submittedPassword?: string;
+  submittedConfirmPassword?: string;
+}
+
+export const FormFields = ({
+  register,
+  errors,
+  watch,
+  submittedPassword = '',
+  submittedConfirmPassword = '',
+}: FormFieldsProps) => {
+  const passwordValue = watch?.('password') || '';
+  const confirmPasswordValue = watch?.('confirmPassword') || '';
+  const displayPassword = register ? passwordValue : submittedPassword;
+  const displayConfirmPassword = register
+    ? confirmPasswordValue
+    : submittedConfirmPassword;
+  const [showPassword, setShowPassword] = useState(false);
+
+  const allCountries = useAppSelector((state) => state.country.countries);
+
+  const passwordStrength = getPasswordStrength(displayPassword);
+  const strengthLabels = [
+    'Very Weak',
+    'Weak',
+    'Fair',
+    'Good',
+    'Strong',
+    'Very Strong',
+  ];
+  const strengthColors = [
+    '#dc3545',
+    '#fd7e14',
+    '#ffc107',
+    '#a0d911',
+    '#52c41a',
+    '#1890ff',
+  ];
```

Comment:
These are constants, so it’s better to move them outside the component function to avoid recreating them on every render.
This improves performance and keeps the component cleaner.